gson/pom.xml

```diff
@@ -365,5 +365,27 @@
         <excludeTestCompilation />
       </properties>
     </profile>
+    <!-- For JDK 25+ skip tests which rely on ProGuard because it does not support JDK 25 yet -->
+    <!-- TODO: Remove this once ProGuard supports JDK 25, and there is a solution for https://github.com/Guardsquare/proguard/issues/473 -->
+    <profile>
+      <id>JDK25</id>
+      <activation>
+        <jdk>[25,)</jdk>
+      </activation>
+      <properties>
+        <excludeTestCompilation>com/google/gson/functional/EnumWithObfuscatedTest.java</excludeTestCompilation>
```

Comment:
For simplicity I reused the existing `excludeTestCompilation` property which we are already using for excluding JDK 17+ tests on JDK 11.

A slightly cleaner solution might be to adjust the maven-surefire-plugin configuration instead, to exclude that test. But I am not sure if that is worth it, given that hopefully soon in the future ProGuard supports JDK 25.